pkg/storage/stores/tsdb/store.go

```diff
@@ -17,37 +19,69 @@ import (
 	util_log "github.com/grafana/loki/pkg/util/log"
 )
 
-// we do not need to build store for each schema config since we do not do any schema specific handling yet.
-// If we do need to do schema specific handling, it would be a good idea to abstract away the handling since
-// running multiple head managers would be complicated and wasteful.
-var storeInstance *store
-
 type store struct {
-	indexWriter IndexWriter
-	indexStore  series.IndexStore
+	indexShipper indexshipper.IndexShipper
+	indexWriter  IndexWriter
+	indexStore   series.IndexStore
+	stopOnce     sync.Once
 }
 
+type newStoreFactoryFunc func(
```

Comment:
Is this used anywhere else than for `NewStore`?

Reply:
No, its not. It is just for defining function signature.